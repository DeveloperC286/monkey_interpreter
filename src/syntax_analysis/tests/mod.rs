use rstest::rstest;

#[rstest(
    code,
    snapshot_name,
    case("", "test_syntax_analysis_tokens_input_edgecases_case1"),
    case(";", "test_syntax_analysis_tokens_input_edgecases_case2")
)]
fn test_syntax_analysis_tokens_input_edgecases(code: &str, snapshot_name: &str) {
    assert_expected_returned_abstract_syntax_tree!(code, snapshot_name)
}
