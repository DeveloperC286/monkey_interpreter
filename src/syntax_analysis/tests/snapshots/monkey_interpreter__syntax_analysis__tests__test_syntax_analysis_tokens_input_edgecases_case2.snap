---
source: src/syntax_analysis/tests/mod.rs
expression: "crate::syntax_analysis::SyntaxAnalysis::from(crate::lexical_analysis::LexicalAnalysis::from(code))"

---
AbstractSyntaxTree {
    abstract_syntax_tree: [],
    syntax_parsing_errors: [
        "Syntax error : Do not know how to parse SemiColon as an expression.",
    ],
}
